File: Basic Coding Problems/01 Fundamentals/001 Write a program to perform input-output of all basic data types/C/input_output.c
#include <stdio.h>

int main() {
    char character;
    unsigned char unsigned_character;
    short short_integer;
    int integer;
    unsigned int unsigned_integer;
    long long_integer;
    unsigned long unsigned_long;
    unsigned long long unsigned_long_long;
    float float_value;
    double double_value;
    long double long_double_value;

    printf("Enter a character: ");
    character = getchar();
    getchar();

    printf("Enter an unsigned character: ");
    unsigned_character = getchar();
    getchar();

    printf("Enter a short integer: ");
    scanf("%hd", &short_integer);

    printf("Enter an integer: ");
    scanf("%d", &integer);

    printf("Enter an unsigned integer: ");
    scanf("%u", &unsigned_integer);

    printf("Enter a long integer: ");
    scanf("%ld", &long_integer);

    printf("Enter an unsigned long integer: ");
    scanf("%lu", &unsigned_long);

    printf("Enter an unsigned long long integer: ");
    scanf("%llu", &unsigned_long_long);

    printf("Enter a float: ");
    scanf("%f", &float_value);

    printf("Enter a double: ");
    scanf("%lf", &double_value);

    printf("Enter a long double: ");
    scanf("%Lf", &long_double_value);

    printf("\nThe character is: %c\n", character);
    printf("The unsigned character is: %c\n", unsigned_character);
    printf("The short integer is: %hd\n", short_integer);
    printf("The integer is: %d\n", integer);
    printf("The unsigned integer is: %u\n", unsigned_integer);
    printf("The long integer is: %ld\n", long_integer);
    printf("The unsigned long integer is: %lu\n", unsigned_long);
    printf("The unsigned long long integer is: %llu\n", unsigned_long_long);
    printf("The float is: %f\n", float_value);
    printf("The double is: %lf\n", double_value);
    printf("The long double is: %Lf\n", long_double_value);

    return 0;
}

File: Basic Coding Problems/01 Fundamentals/001 Write a program to perform input-output of all basic data types/Java/input_output.java
import java.util.Scanner;

public class input_output {
    public static void main(String[] args) {
        byte byteValue;
        char character;
        short shortInteger;
        int integer;
        long longInteger;
        float floatValue;
        double doubleValue;

        // Get user input
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a byte: ");
        byteValue = scanner.nextByte();

        System.out.print("Enter a character: ");
        character = scanner.next().charAt(0);

        System.out.print("Enter a short integer: ");
        shortInteger = scanner.nextShort();

        System.out.print("Enter an integer: ");
        integer = scanner.nextInt();

        System.out.print("Enter a long integer: ");
        longInteger = scanner.nextLong();

        System.out.print("Enter a float: ");
        floatValue = scanner.nextFloat();

        System.out.print("Enter a double: ");
        doubleValue = scanner.nextDouble();

        // Print the results
        System.out.println("\nThe byte is: " + byteValue);
        System.out.println("\nThe character is: " + character);
        System.out.println("The short integer is: " + shortInteger);
        System.out.println("The integer is: " + integer);
        System.out.println("The long integer is: " + longInteger);
        System.out.println("The float is: " + floatValue);
        System.out.println("The double is: " + doubleValue);
    }
}